src/Momento.Sdk/Auth/EnvironmentTokenProvider.cs

```diff
@@ -0,0 +1,34 @@
+namespace Momento.Sdk.Auth;
+
+using System;
+using Momento.Sdk.Exceptions;
+using Momento.Sdk.Internal;
+
+public class EnvironmentTokenProvider : ICredentialProvider
+{
+    public string AuthToken { get; private set; }
+    public string ControlEndpoint {get; private set; }
+    public string CacheEndpoint { get; private set; }
+
+    /// <summary>
+    /// Reads and parses a JWT token stored as an environment variable.
+    /// <param name="name">Name of the environment variable that contains the JWT token.</param>
+    /// </summary>
+    public EnvironmentTokenProvider(string name)
```

Comment:
My only slight concern here is around the naming. When we do introduce new tokens, users might still wan to be able to set them via an environment variable. However, they'll contain something different in the claims (won't have the control and cache endpoints), so the rest of the logic in this provider won't work.

I'm not sure we have great names for these current tokens vs. new tokens (I've sometimes called current ones "legacy" but uh... that's not a good name when we haven't released new tokens 😅) - new tokens might be called "API tokens" and "session tokens" (there will be two different kinds, neither of them will have this info in claims). Maybe something like "Momento auth token" for the current ones, so maybe this should be something like `EnvironmentMomentoAuthTokenProvider` - that's kind of a mouthful though and not sure it's actually helpful. Naming is the worst.

Reply:
FWIW a `APITokenProvider` would probably look something like (sorry I don't really know c#)

```
public APITokenProvider(string apiToken, string endpoint)
     AuthToken = apiToken;
     ControlEndpoint = $"control.{endpoint}"
     CacheEndpoint = $"cache.{endpoint}"
```

Just in case that extra context helps.

I wonder if it would be easiest for now just to have like a `MomentoAuthTokenProvider` that doesn't do any of the environment variable reading? Idk.

Reply:
@rlinehan Since these are specifically JWT tokens, what would you think about `EnvironmentJwtTokenProvider`? The environment variable reading is just an implementation detail of the specific class, and the interface can support the `APITokenProvider` you described just fine.

Reply:
Hmmm the new tokens will also be JWTs though, so I'm not quite sure that works. 🤔